packages/react/src/hooks/useSessionMessages.ts

```diff
@@ -124,13 +144,25 @@ export function useSessionMessages(session?: UseSessionReturn): UseSessionMessag
     }
   }, [receivedMessages]);
 
+  const sortedReceivedMessages = React.useMemo(() => {
+    return receivedMessages.sort((a, b) => {
+      const aFirstReceivedAt = messageFirstReceivedTimeMap.get(a.id);
+      const bFirstReceivedAt = messageFirstReceivedTimeMap.get(b.id);
+      if (typeof aFirstReceivedAt === 'undefined' || typeof bFirstReceivedAt === 'undefined') {
+        return 0;
+      }
+
+      return aFirstReceivedAt.getTime() - bFirstReceivedAt.getTime();
+    });
+  }, [receivedMessages, messageFirstReceivedTimeMap]);
```

Comment:
issue: because `receivedMessages` is in a memo and `messageFirstReceivedTimeMap` is set in a useEffect that depends on `receivedMessages` every update will currently trigger sorting to run twice